routes/api.rb

```diff
@@ -1,35 +1,39 @@
 module ChistApp
   class ApiRoutes < Cuba
     define do
-      on authenticated_request do
-        on post do
-          on 'chists' do
-            attrs = chist_params_from_request
+      unauthorized! unless authenticated_request
 
-            ctx = ChistApp::Context::ChistCreation.new(attrs["chist"], self)
+      on post do
+        on 'chists' do
+          attrs = chist_params_from_request
 
-            result = ctx.call
+          ctx = ChistApp::Context::ChistCreation.new(attrs, self)
 
-            if result == :success
-              body = { :chist => {
-                      :url => "#{ENV["SITE_URL"]}/chists/#{ctx.chist.id}",
-                      :created_at => Time.now } }
+          result = ctx.call
 
-              headers = { "Content-type" => "application/json",
-                          "Location" => body[:chist][:url] }
+          if result == :success
+            body = { :chist => ctx.chist.to_hash.merge(:url => "#{ENV["SITE_URL"]}/#{ctx.chist.url}") }
 
-              halt [201, headers, body.to_json]
-            else
-              res.status = 500
-              res.write("Internal Server Error")
-              halt res.finish
-            end
+            headers = { "Location" => body[:chist][:url] }
```

Comment:
Yes, it's a good idea. So far the API is growing on demand, and I just added this route for the Firefox add-on I'm building.
As we currently don't have the API route for individual chists, I'd leave it as it is, with the link to the browser version, and discuss the individual show route in a different thread so we can specify the format properly.
The route for the raw version is also a good idea, we might use a parameter instead of a separate route, but let's define that separately.